lib/remote.js

```diff
@@ -83,71 +62,13 @@ module.exports = (cfg, names, params, callback) => {
   headers['X-Faaslang'] = 'true';
   body = Buffer.from(JSON.stringify(params));
 
-  cfg.token && (headers['Authorization'] = `Bearer ${cfg.token}`);
-  cfg.keys && (headers['X-Authorization-Keys'] = JSON.stringify(cfg.keys));
-  cfg.webhook && (headers['X-Webhook'] = cfg.webhook); // TODO: Deprecate
-  cfg.convert && (headers['X-Convert-Strings'] = 'true');
-  headers['X-Stdlib-Stack-Trace'] = process.env.__STDLIB_STACK_TRACE || '';
-  cfg.bg && (pathname += `:bg${typeof cfg.bg === 'string' ? '=' + encodeURIComponent(cfg.bg) : ''}`);
-
-  let responded = false;
-
-  let req = (cfg.port === 443 ? https : http).request({
-    host: cfg.host,
-    method: 'POST',
-    headers: headers,
-    port: cfg.port,
-    path: `${cfg.path}${pathname}`,
-    agent: false
-  }, function (res) {
-
-    let buffers = [];
+  return request(cfg, 'POST', pathname, headers, body, (err, res) => {
 
-    res.on('data', chunk => buffers.push(chunk));
-    res.on('end', () => {
-
-      var response = Buffer.concat(buffers);
-      var contentType = (res.headers['content-type'] || '').split(';')[0];
-
-      if (contentType === 'application/json') {
-        response = response.toString();
-        try {
-          response = JSON.parse(response);
-        } catch(e) {
-          response = null;
-        }
-      } else if (contentType === 'text/plain') {
-        response = response.toString();
-      }
-
-      response = parseBuffers(response, contentType);
-      responded = true;
-
-      if (((res.statusCode / 100) | 0) !== 2) {
-        let message = typeof response === 'object' ?
-          (response && response.error && response.error.message) || ('Unspecified error running remote Standard Library function "' + names.join('.') + '"') :
-          response;
-          let error = new Error(message);
-          if (response.error && response.error.hasOwnProperty('details')) {
-            error.details = response.error.details;
-          }
-          if (response.error && response.error.hasOwnProperty('type')) {
-            error.type = response.error.type;
-          }
-          if (response.error && response.error.hasOwnProperty('stack')) {
-            error.stack = response.error.stack;
-          }
-          return callback(error, response);
-      } else {
-        return callback(null, response, res.headers);
```

Comment:
It looks like we return `res.headers` here too -- we should not change this behavior